src/SharpCompress/Readers/AbstractReader.cs

```diff
@@ -181,16 +182,16 @@ public void WriteEntryTo(Stream writableStream)
 
             var streamListener = this as IReaderExtractionListener;
             streamListener.FireEntryExtractionBegin(Entry);
-            Write(writableStream);
+            Write(writableStream, streamListener);
             streamListener.FireEntryExtractionEnd(Entry);
             wroteCurrentEntry = true;
         }
 
-        internal void Write(Stream writeStream)
+        internal void Write(Stream writeStream, IReaderExtractionListener streamListener)
         {
             using (Stream s = OpenEntryStream())
             {
-                s.TransferTo(writeStream);
+                s.TransferTo(writeStream, (sizeTransferred, iterations) => streamListener.FireEntryExtractionProgress(Entry, sizeTransferred, iterations));
```

Comment:
I guess this is okay. I wouldn't have a problem just duplicating the transfer code to have listening and not listening.

Kind of doesn't make sense to have a lambda just to call the listener when the listener can just be passed in.